internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -1474,10 +1504,26 @@ func resourceOrchestratedVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, m
 				d.Set("extensions_time_budget", extensionsTimeBudget)
 
 				encryptionAtHostEnabled := false
-				if profile.SecurityProfile != nil && profile.SecurityProfile.EncryptionAtHost != nil {
-					encryptionAtHostEnabled = *profile.SecurityProfile.EncryptionAtHost
+				vtpmEnabled := false
+				secureBootEnabled := false
+
+				if secprofile := profile.SecurityProfile; secprofile != nil {
+					if secprofile.EncryptionAtHost != nil {
+						encryptionAtHostEnabled = *secprofile.EncryptionAtHost
+					}
```

Comment:
could we simplify this to use `pointer.From`? Previously we would have nil checked these, but going forward we prefer to `pointer.From` to handle these

```suggestion
						encryptionAtHostEnabled = pointer.From(secprofile.EncryptionAtHost)
					
```